middlewares/validators/authValidator.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable newline-per-chained-call */
+// const emailExist = require('../../lib/utils/checkers/emailExistChecker');
+// const userModelManager = require('../../lib/modelManagers/usermodel');
+
+const errorMessage = (res, message) => res.status(400).json({
+  status: false,
+  message
+});
+
+/**
+   * @description check against errors on signup fields
+   * @param {object} req
+   * @param {object} res
+   * @param {object} next
+   * @returns {object} returns error object if error exist
+   */
+const authValidator = (req, res, next) => {
+  req.check('email')
+    .notEmpty().withMessage('Email is required')
+    .isEmail().trim().withMessage('Email must be valid');
+  req.check('password')
+    .notEmpty().withMessage('Password is required')
+    .isLength({ min: 8 }).withMessage('Password must not be less than 8 characters')
+    .matches('[a-z]').withMessage('password must contain a lowercase')
+    .matches('[A-Z]').withMessage('password must contain a uppercase')
+    .matches('[0-9]').withMessage('password must contain a number');
+  req.check('firstname')
+    .notEmpty().withMessage('firstname is required')
+    .isAlpha().trim().withMessage('firstname must be alphabet letters');
+  req.check('lastname')
+    .notEmpty().withMessage('lastname is required')
+    .isAlpha().trim().withMessage('lastname must be alphabet letters');
+
```

Comment:
`req.check('firstname')
    .notEmpty().withMessage('firstname is required')
    .isAlpha().trim().withMessage('firstname must be alphabet letters');` is similar to 

`
req.check('lastname')
    .notEmpty().withMessage('lastname is required')
    .isAlpha().trim().withMessage('lastname must be alphabet letters');
`
consider abstracting it into a function. The function should look like the function below



```js

   function validateName(req, name) {
         req.check(name')
             .notEmpty().withMessage(`${name} is required`)
             .isAlpha().trim().withMessage(`${name} must be alphabet letters`);
       }  
```

Reply:
fixed